lib/block_view.go

```diff
@@ -276,6 +276,21 @@ type RecloutEntry struct {
 	isDeleted bool
 }
 
+// PinEntry stores the timestamp and state of a PostEntry.
+// It's used in the UtxoView to keep track of pinned posts not yet in the database.
+type PinEntry struct {
```

Comment:
High-level thoughts after reading through.

It would be good to understand what the value of a PinEntry is. My view on pinned posts is we'd implement them the exact same way as we implement IsHidden on posts, which is brain-dead simple and doesn't even require any change to the disconnect logic.

Then, to get an index of the pinned posts, we can modify PutPostEntryMappings and DeletePostEntryMappings in db_utils.go to simply update a separate little index of <PublicKey, Timestamp> for the pinned posts. We could even include *another* field in addition to IsPinned on the PostEntry, which is PinTimestamp, which can allow the user to order the pinned posts on their profile by pinning them in a different order. This would also require no changes to the disconnect logic I think.

I may be missing the reason why PinEntry is required, and if that's the case please let me know!

Reply:
I think the reason @bluepartyhat added the PinEntry due to the complications that would arise in `DBPutPostEntryMappingsWithTxn`. Unlike `isHidden`, `isPinned` results in different indexing. So when we flush to the DB, we need to know the previous pin status of the PostEntry (which is not retained on the PostEntry) so we can delete it in `DBDeletePostEntryMappingsWithTxn` and can then be correctly re-added in `DBPutPostEntryMappingsWithTxn`.  I think adding the complication there would probably be preferable to adding the PinEntry, but let's chat about this since the indexing is odd.